[docs] Formatting + Inform updaters about audit

CWSpear opened the pull request:
Just tweaked some formatting to hopefully increase readability and added a note about running `brew audit...` under the update section.

mikemcquaid commented:
@CWSpear Would you mind splitting the formatting changes into another PR or at least commit? As-is it's very hard to review what the actual changes are.

CWSpear commented:
@mikemcquaid I split them up, but it's still hard to see the diff as it's not highlighting the added part:

I added **run `brew audit --online --strict foo` (fix any issues if necessary until it passes),** between **`brew install foo`,** and **`git commit`**.

My first contribution to Homebrew was a PR to bump a version of a package, and I didn't know about `brew audit` and could have saved some time if I'd run it. (I just went off of the `CONTRIBUTING.md` hoping everything I needed would be there). So I was just trying to add a note about it there.

mikemcquaid commented:
That's more helpful now. That's good to have but I'd say just `run brew audit foo and fix any issues`; we don't need `--strict --online` for version bumps.

mikemcquaid commented:
@CWSpear A few nits and comments but I'm liking this in general :+1:

CWSpear commented:
@mikemcquaid actually, while we're at it, would you mind if we refactored so that each subsection is a bulleted list instead of a comma delineated list?

> run `brew update` (twice), run and read `brew doctor`, read [the Troubleshooting Checklist](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting), open an issue on the formula's repository.

Becomes:

> * run `brew update` (twice)
> * run and read `brew doctor`
> * read [the Troubleshooting Checklist](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting)
> * open an issue on the formula's repository

etc, etc. Could even be numbered if you think that's better

CWSpear commented:
@mikemcquaid See new commits. How does that look?

mikemcquaid commented:
@CWSpear Sorry about the delay: could you squash to two commits for review: style changes and content changes?

CWSpear commented:
I can, but it won't really help as I said before. Because of the paragraphs, it doesn't really show a diff of what's changed and it's not very helpful. I tried to detail everything I changed content-wise (which was very little) and put it in commits that made it as clear as possible.

mikemcquaid commented on CONTRIBUTING.md:
```diff
@@ -1,24 +1,41 @@
-- [Email security@brew.sh](mailto:security@brew.sh) (a private mailing list) with detailed reproduction instructions using [our PGP key](https://keybase.io/homebrew/key.asc).
+* [Email security@brew.sh](mailto:security@brew.sh) (a private mailing list) with detailed reproduction instructions using [our PGP key](https://keybase.io/homebrew/key.asc)
+
+---
```
Can remove this line.

CWSpear replied:
Removed

mikemcquaid commented:
Looks good to me. I'm wondering if it's easier/harder to scan compared to how it was before. @Homebrew/maintainers any thoughts before I merge this?

UniqMartin commented:
Personally, I find the reformatted version easier to scan, even if it makes the whole document longer.

apjanke commented:
I much prefer the new reformatted version. Way easier to scan for me, and probably easier to use as a checklist if you're actually going through one of these processes.

apjanke commented on CONTRIBUTING.md:
```diff
@@ -1,24 +1,39 @@
+### Add a new formula for `foo` version `2.3.4` from `$URL`
 
-- Read [the Formula Cookbook](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#formula-cookbook) or: `brew create $URL` and make edits, `brew install foo`, `brew audit --online --strict foo`, `git commit` with message formatted `foo 2.3.4 (new formula)`, [open a pull request](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-H ...
+* Read [the Formula Cookbook](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#formula-cookbook) or: `brew create $URL` and make edits
```
Minor nit: should probably be "read" for consistency with the other list items, which are not in initial caps. And it may flow better without the colon after "or", which is usually used to introduce longer lists.

apjanke commented on CONTRIBUTING.md:
```diff
@@ -1,24 +1,39 @@
+### Submit a `1.2.3` version upgrade for the `foo` formula
 
-- `brew edit foo`, edit [`url`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method) and [`sha256`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#sha256%3D-class_method)/[`tag`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method), leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#bottle ...
+* `brew edit foo`,
```
spurious trailing comma?

apjanke commented on CONTRIBUTING.md:
```diff
@@ -1,24 +1,39 @@
 
-- `brew edit foo`, edit [`url`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method) and [`sha256`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#sha256%3D-class_method)/[`tag`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method), leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#bottle ...
+* `brew edit foo`,
+* edit [`url`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method) and [`sha256`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#sha256%3D-class_method)/[`tag`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#url-class_method), leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#bottle-class_method) as ...
```
spurious trailing comma?

apjanke commented on CONTRIBUTING.md:
```diff
@@ -1,24 +1,39 @@
-- `brew edit foo` and make edits, leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#bottle-class_method) as-is, `brew install foo`, `git commit` with message formatted `foo: fix <insert details>.`, [open a pull request](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-hom ...
+* `brew edit foo` and make edits
+* leave the [`bottle`](http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula#bottle-class_method) as-is
+* `brew install foo`, `git commit` with message formatted `foo: fix <insert details>`
```
Maybe split in to two items and add note about testing? Should test/audit on each change, not just new formulae.

* `brew install foo`, `brew test foo`, and `brew audit --online --strict foo`
* `git commit` with message formatted `foo: fix <insert details>

CWSpear replied:
I left `--online --strict` out, since @mikemcquaid said

> we don't need --strict --online for version bumps.

so I assumed that'd apply for fixes, too?

CWSpear commented:
@apjanke one comment about the `audit` flags, but otherwise made your changes. Good catches on the commas and stuff: left over artifacts from converting them from comma separated lists.

mikemcquaid commented:
I'm sold on merging this then, thanks for the feedback @UniqMartin, @apjanke.

@CWSpear This has a merge conflict :sob: but after that's resolved I'm :+1: to merge this as-is.

CWSpear commented:
Yes, thanks all for the feedback!

The PR has been rebased against master.

You can see what conflicted here: https://github.com/Homebrew/homebrew/commit/332430d433ae77d18bec8eb5aa9f61ca9d7cb8da

I added that to this PR.

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

CWSpear commented:
=]

UniqMartin commented:
Awesome! :clap:

apjanke commented:
:+1: Thanks!
